plenum/test/client/test_protocol_version.py

```diff
@@ -15,22 +19,45 @@ def request_num(request):
     return int(request.param)
 
 
-def test_client_send_incorrect_ledger_status(looper, txnPoolNodeSet,
-                                             sdk_pool_handle,
-                                             sdk_wallet_client):
-    # SDK client send LEDGER_STATUS without protocoloVersion.
-    # Node discard this message and send client's LEDGER_STATUS back
-    sdk_send_random_and_check(looper, txnPoolNodeSet,
-                              sdk_pool_handle, sdk_wallet_client, 1)
+def test_client_send_incorrect_ledger_status(looper, txnPoolNodeSet):
+    # Client send LEDGER_STATUS without protocoloVersion field.
+    # Node send REQNACK and client's LEDGER_STATUS back
     spy = txnPoolNodeSet[0].spylog
 
+    # msg without protocolVersion field
+    msg = {'op': 'LEDGER_STATUS', f.LEDGER_ID.nm: 0, f.TXN_SEQ_NO.nm: 4, f.PP_SEQ_NO.nm: None, f.VIEW_NO.nm: None,
+           f.MERKLE_ROOT.nm: '31ftJ1dwFmtVhdPqKo55BYX9bYxezNRioxMvPoZTjKZ5'}
+    sender = 'client_1'
+    wrappedMsg = (msg, sender)
+
+    # emulate client sends LEDGER_STATUS
+    txnPoolNodeSet[0].handleOneClientMsg(wrappedMsg)
+
+    # node discarded message
     discards = spy.getAll(Node.discard)
-    output = discards[0][3]
-    assert output['msg'][0]['op'] == 'LEDGER_STATUS'
-    assert 'missed fields - protocolVersion. ' + error_msg in output['reason']
+    flag = False
+    for discrad in discards:
+        if discrad[3]['msg'][0] == msg:
```

Comment:
What does `3` mean here?

Reply:
That is how spylog structured. 3 index contains params of called method

Reply:
changed to be more objective